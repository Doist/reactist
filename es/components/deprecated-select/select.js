import {
    objectWithoutProperties as _objectWithoutProperties,
    objectSpread2 as _objectSpread2,
} from '../../_virtual/_rollupPluginBabelHelpers.js'
import * as React from 'react'
import classNames from 'classnames'

const _excluded = ['value', 'options', 'onChange', 'disabled', 'className', 'defaultValue']

function Select(_ref) {
    let { value, options = [], onChange, disabled = true, className = '', defaultValue } = _ref,
        otherProps = _objectWithoutProperties(_ref, _excluded)

    const selectClassName = classNames(
        'reactist_select',
        {
            disabled,
        },
        className,
    )
    return /*#__PURE__*/ React.createElement(
        'select',
        _objectSpread2(
            {
                className: selectClassName,
                value: value,
                onChange: (event) => (onChange ? onChange(event.target.value) : undefined),
                disabled: disabled,
                defaultValue: defaultValue,
            },
            otherProps,
        ),
        options == null
            ? void 0
            : options.map((option) =>
                  /*#__PURE__*/ React.createElement(
                      'option',
                      {
                          key: option.key || option.value,
                          value: option.value,
                          disabled: option.disabled,
                      },
                      option.text,
                  ),
              ),
    )
}

Select.displayName = 'Select'
Select.defaultProps = {
    options: [],
    disabled: false,
}

export { Select }
//# sourceMappingURL=select.js.map
