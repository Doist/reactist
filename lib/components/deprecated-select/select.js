"use strict";Object.defineProperty(exports,"__esModule",{value:!0});var e=require("../../_virtual/_rollupPluginBabelHelpers.js"),t=require("react"),a=require("classnames");function l(e){return e&&"object"==typeof e&&"default"in e?e:{default:e}}function r(e){if(e&&e.__esModule)return e;var t=Object.create(null);return e&&Object.keys(e).forEach((function(a){if("default"!==a){var l=Object.getOwnPropertyDescriptor(e,a);Object.defineProperty(t,a,l.get?l:{enumerable:!0,get:function(){return e[a]}})}})),t.default=e,t}var u=r(t),n=l(a);const o=["value","options","onChange","disabled","className","defaultValue"];function s(t){let{value:a,options:l=[],onChange:r,disabled:s=!0,className:i="",defaultValue:c}=t,d=e.objectWithoutProperties(t,o);const f=n.default("reactist_select",{disabled:s},i);return u.createElement("select",e.objectSpread2({className:f,value:a,onChange:e=>r?r(e.target.value):void 0,disabled:s,defaultValue:c},d),null==l?void 0:l.map(e=>u.createElement("option",{key:e.key||e.value,value:e.value,disabled:e.disabled},e.text)))}s.displayName="Select",s.defaultProps={options:[],disabled:!1},exports.Select=s;
//# sourceMappingURL=select.js.map
