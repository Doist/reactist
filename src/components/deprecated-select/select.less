@import '../styles/constants.less';

.reactist_select {
    .normal_text();
    background-color: white;
    border: 1px solid @color_border_grey;
    border-radius: 3px;

    // nuke the browser styles
    -moz-appearance: none;
    -webkit-appearance: none;
    appearance: none;
    margin: 0;
    padding: 5px 25px 5px 10px;
    background-image: url('data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iOCIgaGVpZ2h0PSIxMCIgdmlld0JveD0iMCAwIDggMTAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPGcgZmlsbD0ibm9uZSIgZmlsbC1ydWxlPSJub256ZXJvIiBzdHJva2U9IiMwMDAiIHN0cm9rZS1saW5lY2FwPSJyb3VuZCIgc3Ryb2tlLWxpbmVqb2luPSJyb3VuZCIgb3BhY2l0eT0iLjgiPgogICAgPHBhdGggZD0iTTEgM2wzLTMgMyAzTTEgN2wzIDMgMy0zIi8+CiAgPC9nPgo8L3N2Zz4K');
    background-position: calc(~'100% - 11px') center;
    background-repeat: no-repeat;
    background-size: 8px 11px;

    &.disabled {
        background-color: @color_grey_11;
    }
    &:focus {
        outline: none;
        border: 1px solid @primary_color;
    }
}
