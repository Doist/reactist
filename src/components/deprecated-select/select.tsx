import React from 'react'
import classNames from 'classnames'

import './select.less'

type Option = {
    /** Optional key for each option. If not provided `value` is used. */
    key?: string | number
    /** Value of the option. */
    value: string | number
    /** Text to display for the option. */
    text?: string | number
    /** Whether the options is disabled or not. */
    disabled?: boolean
}

type Props = {
    className?: string
    disabled?: boolean
    /** Currently selected value. */
    value?: string | number
    /** Callback for the change event. Will be called with the next value (not the full event). */
    onChange?: (value: string) => void
    /** Options that are rendered in the select. */
    options?: Option[]
    /** Value to initially be set */
    defaultValue?: string | number
}

function Select({
    value,
    options = [],
    onChange,
    disabled = true,
    className = '',
    defaultValue,
    ...otherProps
}: Props) {
    const selectClassName = classNames('reactist_select', { disabled }, className)
    return (
        <select
            className={selectClassName}
            value={value}
            onChange={(event) => (onChange ? onChange(event.target.value) : undefined)}
            disabled={disabled}
            defaultValue={defaultValue}
            {...otherProps}
        >
            {options?.map((option) => (
                <option
                    key={option.key || option.value}
                    value={option.value}
                    disabled={option.disabled}
                >
                    {option.text}
                </option>
            ))}
        </select>
    )
}
Select.displayName = 'Select'
Select.defaultProps = {
    options: [],
    disabled: false,
}

export { Select }
