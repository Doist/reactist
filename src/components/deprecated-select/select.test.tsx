import React from 'react'
import { shallow } from 'enzyme'

import { Select } from './select'

describe('Select', () => {
    it('renders without crashing', () => {
        const select = shallow(getSelect())
        expect(select).toMatchSnapshot()
    })

    it('uses key of option when supplied', () => {
        const select = shallow(
            getSelect({
                options: [{ value: 'test', text: 'test', key: 'test-key' }],
            }),
        )
        expect(select.find('option').get(0).key).toBe('test-key')
    })

    it('calls onChange handler with selected value', () => {
        const onChangeSpy = jest.fn()
        const select = shallow(
            getSelect({
                onChange: onChangeSpy,
                options: [{ value: 'test2', text: 'test2' }],
            }),
        )

        select.simulate('change', { target: { value: 'test2' } })
        expect(onChangeSpy).toHaveBeenLastCalledWith('test2')
    })

    // helpers
    function getSelect(props: React.ComponentProps<typeof Select> = {}) {
        return (
            <Select
                onChange={jest.fn()}
                value="test"
                options={[{ value: 'test', text: 'test' }]}
                {...props}
            />
        )
    }
})
