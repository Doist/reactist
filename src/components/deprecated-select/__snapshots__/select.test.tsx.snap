// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders without crashing 1`] = `
<select
  className="reactist_select"
  disabled={false}
  onChange={[Function]}
  value="test"
>
  <option
    key="test"
    value="test"
  >
    test
  </option>
</select>
`;
